aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -355,6 +355,34 @@ static void polygamma_kernel(TensorIterator& iter, int64_t n) {
   }
 }
 
+static void heaviside_tensor_kernel(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "heaviside_tensor_cpu", [&]() {
+    cpu_kernel(iter, [=](scalar_t self, scalar_t val) -> scalar_t {
+       if (self > 0) {
+         return 1.0;
```

Comment:
This will need an update to handle scalar types properly. Same with its dispatch.

Reply:
Addressed!